File: client/src/app/model/book.ts
export class Book {
  constructor(id?: number, name?: string, author?: string, publicationDate?: Date) {
    this.id = id
    this.name = name
    this.author = author
    this.publicationDate = publicationDate
  }

  id: number;
  name: string;
  author: string;
  publicationDate: Date;
}

File: server/src/main/java/com/rest/app/api/json/Book.java
package com.rest.app.api.json;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.util.Date;
import java.util.List;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class Book {
    @JsonProperty(value = "title")
    private String title;

    @JsonProperty(value = "authors")
    private List<String> authors;

    @JsonProperty(value = "issued")
    private Date publicationDate;
}
